src/skin.cpp

```diff
@@ -200,28 +124,25 @@ void SkinInfillAreaComputation::generateSkinAndInfillAreas()
  */
 void SkinInfillAreaComputation::generateSkinAndInfillAreas(SliceLayerPart& part)
 {
-    Polygons original_outline = (wall_line_count < 1) ? part.outline : part.outline.offset(-(wall_line_width_0 + (wall_line_count - 1) * wall_line_width_x));
-
-    // make a copy of the outline which we later intersect and union with the resized skins to ensure the resized skin isn't too large or removed completely.
-    Polygons upskin;
+    //Make a copy of the outline which we later intersect and union with the resized skins to ensure the resized skin isn't too large or removed completely.
+    Polygons top_skin;
     if (top_layer_count > 0)
     {
-        upskin = Polygons(original_outline);
+        top_skin = Polygons(part.inner_area);
```

Comment:
Why not move this logic in the `calculateTopSkin` function and changed the function signature to `Polygons calculateTopSkin(const SliceLayerPart& part)` so you can make it `const` and do:
`const Polygons top_skin = calculateTopSkin(part);`

Reply:
This change would cause a copy of the polygon data. What you're suggesting here would normally always be a copy of the `Polygons` object if it weren't for RVO, return value optimization. RVO is possible as long as the compiler knows that an object will be returned at the moment where it's constructed, e.g. because you're just calling:

    return Polygons(data);

Then it's of course trivial that this `Polygons` will just be returned. But if there is branching code like this:

    Polygons result(data);
    if(condition)
        return Polygons(); //Early out.
    return result;

then RVO can't be applied because the compiler, upon constructing `result`, doesn't know whether that object is going to be returned.

`calculateTopSkin` and `calculateBottomSkin` are such branching functions. There are several early-outs where you'd return the original input polygon. Otherwise you're replacing the input polygon with the result of a `.difference` call.

Reply:
Thanks for the clarification. It is still on my bucket list to refresh my knowledge regarding RVO, move and copy etc.

Reply:
I refreshed my knowledge regarding RVO due to your question :) I had to look this up as well.